inc/Engine/Media/AboveTheFold/Frontend/Controller.php

```diff
@@ -336,4 +345,30 @@ private function is_mobile(): bool {
 			&& $this->options->get( 'do_caching_mobile_files', 0 )
 			&& wp_is_mobile();
 	}
+
+	/**
+	 * The `inject_beacon` function is used to inject a JavaScript beacon into the HTML content.
+	 *
+	 * @param string $html The HTML content where the beacon will be injected.
+	 *
+	 * @return string The modified HTML content with the beacon script injected just before the closing body tag.
+	 */
+	public function inject_beacon( $html ): string {
+		if ( ! $this->filesystem || ! $this->context->is_allowed() ) {
+			return $html;
+		}
```

Comment:
I think the first condition is not needed because we set the filesystem object if it's null so in all cases it'll be there.

The second condition is duplicated because this method `inject_beacon` is called inside the method `lcp` in the same file and there we are calling this is_allowed method so no need to call it again here.

What do u think?

Reply:
Umm that's true. I've removed them.